bpf/lib/nat.h

```diff
@@ -353,6 +353,92 @@ snat_v4_rev_nat_handle_mapping(struct __ctx_buff *ctx,
 		       NAT_PUNT_TO_STACK : DROP_NAT_NO_MAPPING;
 }
 
+static __always_inline int snat_v4_icmp_rewrite_embedded(struct __ctx_buff *ctx,
```

Comment:
Do you think there could be a way to reuse the same implementation for the core rewrite between `snat_v4_rewrite_ingress()` vs. `snat_v4_icmp_rewrite_embedded()`? It seems like as long as you pass the appropriate tuples into the same function, and pull the context to the appropriate offset, then the core function could be identical? Or am I missing something.

Reply:
Ah maybe the checksumming makes this more difficult. :thinking:

Reply:
I was initially trying to reuse one of those ingress/egress functions but they differ in implementation a bit more, we also want to change source addr of inner where for ingress we want to fix the dest.